arex-web-core/src/main/java/com/arextest/web/core/business/util/SchemaUtils.java

```diff
@@ -0,0 +1,99 @@
+package com.arextest.web.core.business.util;
+
+import com.arextest.web.common.LogUtils;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Slf4j
+public class SchemaUtils {
+    public static void mergeMap(Map<String, Object> contract, Map<String, Object> model) {
+        model.forEach((key, value) -> mergeEntry(contract, key, value));
+    }
+
+    public static String mergeJson(String contract, String model) {
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            Map<String, Object> contractMap = contract == null ? new HashMap<>() : objectMapper.readValue(contract, Map.class);
+            if (model == null) return contract;
+            Map<String, Object> modelMap = objectMapper.readValue(model, Map.class);
+            mergeMap(contractMap, modelMap);
+            return objectMapper.writeValueAsString(contractMap);
+        } catch (JsonProcessingException e) {
+            LogUtils.error(LOGGER, "objectMapper readValue failed! contract:{}, model:{}", contract, model);
+        }
+        return null;
+    }
+
+    private static void mergeEntry(Map<String, Object> contract, String key, Object value) {
+        if (value instanceof Map<?, ?>) {
+            Map<String, Object> mapContract = (Map<String, Object>) contract.getOrDefault(key, new HashMap<>());
+            mergeMap(mapContract, (Map<String, Object>) value);
+            contract.put(key, mapContract);
+        } else if (value instanceof List) {
+            List<Object> listContract = (List<Object>) contract.getOrDefault(key, new ArrayList<>());
+            mergeList(listContract, (List<Object>) value);
+            contract.put(key, listContract);
+        } else {
+            if (contract.get(key) == null) {
+                contract.put(key, handlePrimaryItem(value));
+            }
+        }
+    }
+
+    private static void mergeList(List<Object> contract, List<Object> model) {
+        Object contractItem = null;
+        if (!contract.isEmpty()) {
+            contractItem = contract.remove(0);
```

Comment:
Is it possible to lose some unique nodes by deleting the previous element every time?

Reply:
No. Removing operation is to ensur that List<Object> contract only has one item. I removed the only item and then expand it. Finally, it will be added back to the list in line72.